scripts/averaging/create_TEM_files.py

```diff
@@ -50,45 +75,60 @@ def create_TEM_files(adf):
 
-        #Extract baseline years (which may be empty strings if using Obs):
-        syear_baseline = adf.climo_yrs["syear_baseline"]
-        eyear_baseline = adf.climo_yrs["eyear_baseline"]
-
-        case_names.append(base_name)
-        start_years.append(syear_baseline)
-        end_years.append(eyear_baseline)
-        overwrite_tem_cases.append(tem_opts.get("overwrite_tem_base", False))
-    #End if
+        tem_base_loc = adf.get_basic_info('obs_tem_loc')
+        
+        #If path not specified, skip TEM calculation?
+        if tem_base_loc is None:
+            print("\t 'obs_tem_loc' not found in config file, so no TEM files/diagnostics will be generated.")
+            return
+        else:
+            tem_base_loc = Path(tem_base_loc)
+            #Check if TEM directory exists, and if not, then create it:
+            if not tem_base_loc.is_dir():
+                print(f"    {tem_base_loc} not found, making new directory")
+                tem_base_loc.mkdir(parents=True)
+            #End if
 
-    #Set default to h4
-    hist_num = tem_opts.get("hist_num")
-    if hist_num is None:
-        hist_num = "h4"
+    else:
+        if tem_base_loc:
+            cam_hist_locs.append(adf.get_baseline_info("cam_hist_loc", required=True))
+
+            #Set default to h4
+            hist_num = adf.get_baseline_info("tem_hist_str")
+            if hist_num is None:
+                hist_num = "h4"
+
+            #Extract baseline years (which may be empty strings if using Obs):
+            syear_baseline = adf.climo_yrs["syear_baseline"]
+            eyear_baseline = adf.climo_yrs["eyear_baseline"]
+
+            case_names.append(base_name)
+            start_years.append(syear_baseline)
+            end_years.append(eyear_baseline)
+           
+            tem_base_loc = Path(tem_base_loc)
+            #Check if TEM directory exists, and if not, then create it:
+            if not tem_base_loc.is_dir():
+                print(f"    {tem_base_loc} not found, making new directory")
+                tem_base_loc.mkdir(parents=True)
+            #End if
 
-    #Extract TEM file save location
-    output_loc = tem_opts["tem_loc"]
+            tem_locs.append(tem_base_loc)
+            overwrite_tem_cases.append(adf.get_baseline_info("overwrite_tem", False))
 
-    #If path not specified, skip TEM calculation?
-    if output_loc is None:
-        print("\t 'tem_loc' not found in config file, so no TEM files will be generated.")
-        return
-    else:
-        #Notify user that script has started:
-        print("\n  Generating CAM TEM diagnostics files...")
+            #Set default to h4
+            hist_num = adf.get_baseline_info("hist_num")
+            if hist_num is None:
+                hist_num = "h4"
+            hist_nums.append(hist_num)
+        else:
+            print("\t 'cam_tem_loc' not found in 'diag_cam_baseline_climo', so no TEM files/diagnostics will be generated.")
```

Comment:
I might update this print statement to be:

```suggestion
            print("\t 'cam_tem_loc' not found in 'diag_cam_baseline_climo', so no baseline TEM files/diagnostics will be generated.")
```